src/uct/cuda/base/cuda_md.c

```diff
@@ -71,28 +71,45 @@ uct_cuda_base_query_md_resources(uct_component_t *component,
     CUdevice cuda_device;
     ucs_status_t status;
     char device_name[10];
-    int num_gpus;
+    int i, num_gpus;
 
-    status = UCT_CUDA_CALL(UCS_LOG_LEVEL_DIAG, cudaGetDeviceCount, &num_gpus);
+    status = UCT_CUDADRV_FUNC(cuDeviceGetCount(&num_gpus), UCS_LOG_LEVEL_DIAG);
     if ((status != UCS_OK) || (num_gpus == 0)) {
         return uct_md_query_empty_md_resource(resources_p, num_resources_p);
     }
 
-    for (cuda_device = 0; cuda_device < num_gpus; ++cuda_device) {
+    for (i = 0; i < num_gpus; ++i) {
+        status = UCT_CUDADRV_FUNC(cuDeviceGet(&cuda_device, i),
+                                  UCS_LOG_LEVEL_DIAG);
+        if (status != UCS_OK) {
+            continue;
+        }
+
         uct_cuda_base_get_sys_dev(cuda_device, &sys_dev);
-        if (sys_dev != UCS_SYS_DEVICE_ID_UNKNOWN) {
-            ucs_snprintf_safe(device_name, sizeof(device_name), "GPU%d",
-                              cuda_device);
-            status = ucs_topo_sys_device_set_name(sys_dev, device_name,
-                                                  sys_device_priority);
-            ucs_assert_always(status == UCS_OK);
+        if (sys_dev == UCS_SYS_DEVICE_ID_UNKNOWN) {
+            continue;
         }
+
+        ucs_snprintf_safe(device_name, sizeof(device_name), "GPU%d",
+                          cuda_device);
```

Comment:
`i` is an ordinal number. It corresponds to the number in the list of visible devices. `cuDeviceGet` returns in `cuda_device` a device handle. The device handle will be the same for the same device regardless of device visibility.

Reply:
E.g. there are two devices on the system. if we use `i` instead of `cuda_device`, then:
a. If `CUDA_VISIBLE_DEVICES=0,1`, GPU0 will correspond to device 0, and GPU1 - to device 1.
b. If `CUDA_VISIBLE_DEVICES=1,0`, GPU0 will correspond to device 1, and GPU1 - to device 0.

Reply:
ok, seems it is more representative than ordinal number which was used before